single-kernel/geometric_mean.cpp

```diff
@@ -0,0 +1,122 @@
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+#include "common.h"
+
+namespace s = sycl;
+
+
+class GeometricMeanKernel; // kernel forward declaration
+
+
+class GeometricMean {
+protected:
+  size_t size;
+  size_t local_size;
+  int chunkSize;
+  std::vector<s::float16> input;
+  std::vector<float> output;
+
+  BenchmarkArgs args;
+
+
+  PrefetchedBuffer<s::float16, 1> buf_input;
+  PrefetchedBuffer<float, 1> buf_output;
+
+
+  void fillrandom_float(float* arrayPtr, int width, int height, float rangeMin, float rangeMax) {
```

Comment:
I feel like this is the third implementation of something like this in one of the benchmark codes. Maybe we should provide a general utility for this. (Also, it would be nicer if the argument wasn't a plain pointer but rather something like a span)